channel/handlers.go

```diff
@@ -723,6 +727,21 @@ func (server Server) playerMoveInventoryItem(conn mnet.Client, reader mpacket.Re
 	}
 }
 
+func (server Server) playerDropMesos(conn mnet.Client, reader mpacket.Reader) {
+	amount := reader.ReadInt32()
+	plr, err := server.players.getFromConn(conn)
+
+	if err != nil {
+		return
+	}
+
+	err = plr.dropMesos(amount)
```

Comment:
I don't think the player struct should know about the concept of dropping mesos. This handler should deduct the mesos from the player and then create a drop based on the player id and mesos value i.e. the 'flow' should probably be like:
```
|-handler
      |-> player remove mesos
      | -> pool add drop
```

instead of:

```
|- handler
       |- > player remove mesos
               | -> pool add drop
```